core/Url.php

```diff
@@ -916,4 +916,18 @@ public static function addCampaignParametersToMatomoLink(
         $pathAndQueryString = UrlHelper::getPathAndQueryFromUrl($url, $newParams, true);
         return 'https://' . $domain . '/' . $pathAndQueryString;
     }
+
+    /**
+     * Create an external link tag with optional campaign params if link goes to matomo.org
+     */
+    public static function getExternalLinkTag(
+        ?string $url = null,
```

Comment:
allowing an empty url imho doesn't make sense here. The parameter should really be mandatory
```suggestion
        string $url,
```

Reply:
I thought the same, but since `addCampaignParametersToMatomoLink` supported that then I went with it as well. Will adjust.